ui/src/components/Timeline/state.ts

```diff
@@ -199,26 +198,30 @@ export function selectSelectedVertex(state: State): Vertex | null {
 }
 
 export function selectItems(state: State): Array<TimelineItem> {
-  return selectSortedEntities(state).map(
-    (entity) => new TimelineItem(entity, state.layout)
-  );
-}
-
-export function selectSortedEntities(state: State): Array<Entity> {
   return state.entities
-    .filter(
-      (entity): entity is TimelineEntity => entity.getTemporalStart() !== null
-    )
+    .map((entity) => new TimelineItem(entity, state.layout))
     .sort((a, b) => {
-      const aStart = a.getTemporalStart().value;
-      const bStart = b.getTemporalStart().value;
+      const aStart = a.entity.getTemporalStart();
+      const bStart = b.entity.getTemporalStart();
+
+      if (aStart === null && bStart === null) {
+        return 0;
+      }
+
+      if (aStart === null) {
+        return -1;
+      }
+
+      if (bStart === null) {
+        return 1;
+      }
```

Comment:
I feel like this could be better or re-worked.

Reply:
What would you prefer?

Reply:
It's a good question. I didn't like the look of it, but I'm not honestly sure if I can come up with anything better in terms of being more terse, and easy to understand. It wouldn't be hard to make things more terse and reduce the number of if statements. But on reflection I think that the result would be harder for a future develop to understand. I was thinking of three seperate inline statements of the form:

value = 1;
value = (!a && !b) ? 0 : value;
value = (a && !b) ? -1 : value;
value - (!a && b) ? 1 : value; // this shouldn't be required because it's the only option left.
return value;